### PR TITLE
Update moby to containerd and runc 1.0 final rc

### PR DESCRIPTION
**Pull request by crosbymichael**
Getting this up for review, I'll have to change the repo paths one last time before we can merge but CI takes so long to run I wanted to open this now. 

This is the last rc for the runtime spec and runc before a 1.0 release.

### CODE REVIEW

**Comment by vieux** [comment]
@crosbymichael panic in the unit tests

**Comment by crosbymichael** [comment]
@LK4D4 do you know why `vndr` is freaking out about my dep?

File: oci/defaults_linux.go [hunk 1]
```diff
@@ -12,6 +12,25 @@ func iPtr(i int64) *int64        { return &i }
 func u32Ptr(i int64) *uint32     { u := uint32(i); return &u }
```
Comment (tonistiigi) [inline oci/defaults_linux.go]: `sPtr` should be unused now

File: integration-cli/docker_cli_service_logs_test.go [hunk 2]
```diff
@@ -245,41 +245,3 @@ func (s *DockerSwarmSuite) TestServiceLogsTaskLogs(c *check.C) {
...
-	// make sure task has been deployed.
-	waitAndAssert(c, defaultReconciliationTimeout, d.CheckActiveContainerCount, checker.Equals, 1)
-	// and make sure we have all the log lines
-	waitAndAssert(c, defaultReconciliationTimeout, countLogLines(d, name), checker.Equals, 2)
-
-	cmd := d.Command("service", "logs", name)
-	result = icmd.RunCmd(cmd)
-	// for some reason there is carriage return in the output. i think this is
-	// just expected.
-	c.Assert(result, icmd.Matches, icmd.Expected{Out: "out\r\nerr\r\n"})
```
Comment (tonistiigi) [inline integration-cli/docker_cli_service_logs_test.go]: The test should still be meaningful. Just remove carriage return.

File: oci/defaults_linux.go (hunk 1 above)
Comment (crosbymichael) [inline oci/defaults_linux.go]: done

**Comment by vieux** [comment]
@crosbymichael can you rebase ? 

`github.com/opencontainers/selinux ba1aefe8057f1d0cfb8e88d0ec1dc85925ef987d` is already in master (last line of the `vendor.conf`)

That's why you end up with selinux errors with vndr.

**Comment by crosbymichael** [comment]
@vieux thanks

File: integration-cli/docker_cli_service_logs_test.go (hunk 2 above)
Comment (tonistiigi) [inline integration-cli/docker_cli_service_logs_test.go]: could remove the comment as well

**Comment by crosbymichael** [comment]
I'm having to update and sync the deps with runc because some of them are out of date and causing the validation to fail.  The others will be a bump for docker's versions where runc depends on a newer version.

**Comment by crosbymichael** [comment]
Can we turn off this new vendor comparison because runc is a binary dep I don't know why its diffing the vendor files.

**Comment by vdemeester** [comment]
@crosbymichael as long as `vendor.conf` or `vendor/*` files are changed, the `vendor` check is run

**Comment by crosbymichael** [comment]
@vdemeester its comparing things it shouldn't because runc has a vendor.conf that is not used here because its a binary dep and shouldn't matter.

**Comment by thaJeztah** [comment]
All green now @tonistiigi @mlaventure PTAL!

File: daemon/daemon_unix.go [hunk 3]
```diff
@@ -77,7 +77,7 @@ func getMemoryResources(config containertypes.Resources) *specs.Memory {
 		memory.Reservation = &reservation
 	}
 
-	if config.MemorySwap != 0 {
+	if config.MemorySwap > 0 {
```
Comment (cpuguy83) [inline daemon/daemon_unix.go]: This one has been tricky, IIRC.

Seems -1 (previously unlimited) and 0 (unset) are now treated the same.

Comment (mlaventure) [inline daemon/daemon_unix.go]: This routine seems to only be used when creating the container spec initially (not when updating), in which case it's correct. The default value for that limit being `unlimited`.

File: daemon/daemon_unix.go [hunk 4]
```diff
@@ -95,28 +95,29 @@ func getMemoryResources(config containertypes.Resources) *specs.Memory {
...
 }
 
-func getCPUResources(config containertypes.Resources) *specs.CPU {
-	cpu := specs.CPU{}
+func getCPUResources(config containertypes.Resources) (*specs.LinuxCPU, error) {
+	cpu := specs.LinuxCPU{}
 
-	if config.CPUShares != 0 {
+	if config.CPUShares < 0 {
+		return nil, fmt.Errorf("shares: invalid argument")
```
Comment (mlaventure) [inline daemon/daemon_unix.go]: nit: `cpu shares: ...`

**Review by mlaventure — APPROVED** [review]
LGTM

(one small nit)

**Comment by tonistiigi** [comment]
LGTM

**Comment by thaJeztah** [comment]
@mlaventure can you verify if https://github.com/moby/moby/issues/32667 and https://github.com/moby/moby/issues/32699 should be resolved by this?

**Pull request merged**